plugins/zstack.py

```diff
@@ -0,0 +1,376 @@
+            if isinstance(s, (ARStream, SpectrumStream, MonochromatorSettingsStream)):
+                # TODO: instead of hard-coding the list, a way to detect the type
+                # of live image?
+                logging.info("Not showing stream %s, for which the live image is not spatial", s)
+                dlg.addStream(s, index=None)
+            else:
+                dlg.addStream(s)
+        dlg.addButton("Cancel")
+        dlg.addButton("Acquire", self.acquire, face_colour='blue')
+
+        # Update acq time when streams are added/removed
+        dlg.microscope_view.stream_tree.flat.subscribe(self._update_exp_dur, init=True)
+        dlg.hidden_view.stream_tree.flat.subscribe(self._update_exp_dur, init=True)
+        # TODO: update the acquisition time whenever a setting changes
+
+        # TODO: disable "acquire" button if no stream selected
+
+        # TODO: also display the repetition and axis settings for the SPARC streams.
+
+        ans = dlg.ShowModal()
+
+        if ans == 0:
+            logging.info("Acquisition cancelled")
+        elif ans == 1:
+            logging.info("Acquisition completed")
+        else:
+            logging.warning("Got unknown return code %s", ans)
+
+        if dlg:  # If dlg hasn't been destroyed yet
+            dlg.Destroy()
+
+    def initAcquisition(self):
+        # Move the focus to the start z position
+        logging.debug("Preparing Z Stack acquisition. Moving focus to start position")
+        self.old_pos = self.focus.position.value
+        self.focus.moveAbs({'z': self.zstart.value}).result()
+
+    def stepAcquisition(self, i):
+        self.focus.moveRel({'z': self.zstep.value}).result()
+        
+    def exportAcquisition(self, images):
+        logging.debug("Exporting Z Stack to HDF5")
+        exporter = dataio.find_fittest_converter(self.filename.value)
+        exporter.export(self.filename.value, images)
+
+    def completeAcquisition(self):
+        # Mvoe back to start
+        logging.debug("Z Stack acquisition complete. Returning focus to start position")
+        self.focus.moveAbs(self.old_pos).result()
+        
+    def postProcessing(self, images):
+        cubes = [self.constructCube(im) for im in images]
+        return cubes
+
+    def constructCube(self, images):
+        # images is a list of 5 dim data arrays.
+        ret = []
+        for image in images:
+            stack = np.dstack(image)
+            #stack = np.swapaxes(stack, 0, 2)
+            #stack = np.expand_dims(stack, 1)
+            ret.append(stack[0])
+            
+        # Add back metadata
+        metadata3d = copy.copy(self._metadata)
+        # Extend pixel size to 3D
+        ps_x, ps_y = metadata3d[model.MD_PIXEL_SIZE]
+        ps_z = self.zstep.value
+        metadata3d[model.MD_PIXEL_SIZE] = (ps_x, ps_y, ps_z)
+        metadata3d[model.MD_DIMS] = "ZYX"
+        
+        ret = DataArray(ret, metadata3d)
+            
+        return ret
+
+    # ALL GENERIC
```

Comment:
The comment is a little too generic ;-) Please, extend the comment to be clearer. It should say it's "generic" for any extra dimension acquisition, and list the different methods called, and when they are called.